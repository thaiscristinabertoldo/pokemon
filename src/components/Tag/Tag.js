import React, { memo } from 'react'

import TagStyled from './tag.css'

export const Tag = memo(({ text }) => {
  return <TagStyled>{text}</TagStyled>
})
