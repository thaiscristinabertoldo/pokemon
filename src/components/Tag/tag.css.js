import styled from 'styled-components'

import { COLORS } from '../../constants'

const TagStyled = styled.div`
  border: 1px solid ${COLORS.lightGrey};
  border-radius: 16px;
  color: ${COLORS.mediumGrey};
  font-weight: bold;
  padding: 8px;
  margin-right: 10px;
`

export default TagStyled
