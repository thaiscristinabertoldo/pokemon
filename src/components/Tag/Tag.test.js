import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Tag } from './Tag'

afterEach(cleanup)

describe('Tag', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Tag />)

    expect(asFragment(<Tag />)).toMatchSnapshot()
  })

  it('should take a tag text', () => {
    const { getByText } = render(<Tag text="Tag text" />)

    expect(getByText('Tag text')).toBeTruthy()
  })
})
