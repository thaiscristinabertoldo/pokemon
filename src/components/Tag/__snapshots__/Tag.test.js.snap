// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm kfxWr"
  />
</DocumentFragment>
`;
